src/main/java/org/retest/rebazer/service/RebaseService.java

```diff
@@ -81,32 +106,33 @@ private static Git tryToOpenExistingRepoAndCheckRemote( final File repoFolder, f
 	}
 
 	@SneakyThrows
-	private static boolean originIsRepoUrl( final Git repo, final String expectedRepoUrl ) {
-		return repo.remoteList().call().stream().anyMatch( r -> r.getName().equals( "origin" )
-				&& r.getURIs().stream().anyMatch( url -> url.toString().equals( expectedRepoUrl ) ) );
+	private static boolean originIsRepoUrl( final Git localRepo, final String repoUrl ) {
+		return localRepo.remoteList().call().stream().anyMatch( r -> r.getName().equals( "origin" )
```

Comment:
Have small steps for each operation. For example:
```
return localRepo.remoteList().call().stream() //
	.filter( r -> r.getName().equals( "origin" ) ) //
	.anyMatch( r-> {
		return r.getURIs().stream() //
				.map( URIish::toString ) //
				.anyMatch( repoUrl::equals );
	} );
```

Reply:
Slightly other implementation but fixed